app/src/test/java/com/kickstarter/ui/activities/compose/PreLaunchProjectPageScreenTest.kt

```diff
@@ -0,0 +1,100 @@
+import com.kickstarter.models.Urls
+import org.junit.Test
+
+class PreLaunchProjectPageScreenTest : KSRobolectricTestCase() {
+
+    private val projectImage = composeTestRule.onNodeWithTag("Project Image")
+    private val comingSoonBadge = composeTestRule.onNodeWithTag("Coming soon badge")
+    private val projectName = composeTestRule.onNodeWithTag("Project name")
+    private val projectCreatorLayout = composeTestRule.onNodeWithTag("Creator Layout")
+    private val projectDescription = composeTestRule.onNodeWithTag("Project description")
+    private val projectCategory = composeTestRule.onNodeWithTag("Project category name")
+    private val projectLocation = composeTestRule.onNodeWithTag("Project location name")
+    private val projectSaveButton = composeTestRule.onNodeWithTag("Project save Button")
+    private val projectFollowers = composeTestRule.onNodeWithTag("Project followers")
+
+    @Test
+    fun verifyInitState() {
+        composeTestRule.setContent { // setting our composable as content for test
+            MaterialTheme {
+                val projectState = remember { mutableStateOf(null) }
+                PreLaunchProjectPageScreen(projectState)
+            }
+        }
+        projectImage.assertExists()
+        comingSoonBadge.assertExists()
+        projectName.assertTextEquals("")
+        projectName.assertIsNotDisplayed()
+        projectCreatorLayout.assertExists()
+        projectDescription.assertDoesNotExist()
+        projectCategory.assertDoesNotExist()
+        projectLocation.assertDoesNotExist()
+        projectSaveButton.assertExists()
+        projectSaveButton.assertTextEquals("Notify me on launch")
+        projectFollowers.assertDoesNotExist()
+    }
+
+    @Test
+    fun verifyOnSaveButtonClicked() {
+        val project = ProjectFactory.project().toBuilder()
+            .launchedAt(
+                DateTimeAdapter().decode(CustomTypeValue.fromRawValue(0))
+            )
+            .displayPrelaunch(true)
+            .watchesCount(1)
+            .urls(Urls.builder().build()).build()
+
+        composeTestRule.setContent { // setting our composable as content for test
+            MaterialTheme {
+                val projectState = remember { mutableStateOf(project) }
+
+                PreLaunchProjectPageScreen(
+                    projectState,
+                    onButtonClicked = {
+                        projectState.value = project.toBuilder().isStarred(true).watchesCount(2).build()
+                    },
+                    numberOfFollowers = environment().ksString()?.format(
+                        context().getString(R.string.activity_followers),
+                        "number_of_followers",
+                        projectState.value.watchesCount().toString()
+                    )
+                )
+            }
+        }
+
+        projectImage.assertExists()
+
+        comingSoonBadge.assertExists()
+        comingSoonBadge.assertTextEquals("Coming soon")
```

Comment:
Same for all of this asserts, could they be the resource ones? for example `R.strings.Coming_soon`